components/document/document_view.hpp

```diff
@@ -14,6 +14,7 @@ namespace components::document {
 
         document_view_t();
         explicit document_view_t(document_ptr document);
+        explicit document_view_t(document_view_t&& doc_view);
```

Comment:
Do we really need r-value ref here?

Reply:
Looks like `document_view_t` is a wrapper around `document_ptr` - so might be we can have the same approach as for  `explicit document_view_t(document_ptr document);`?

Reply:
Also it might be important to mark it as `noexcept` https://stackoverflow.com/questions/28627348/noexcept-and-copy-move-constructors

Reply:
So...
Cursor returns const ref to the document_view_t, but if we want to access array or dictionary inside document, it returns new document_view_t as r-value.
And a way to pass it to C# is complicated (You can not pass a class in either direction, but you can access it via pointer). So I store document_view in dynamic memory (otherwise it will go out of scope before C# will get pointer) and pass a pointer, but there is a fork:
1) create move constructor and move created document_view_t to dynamic memory
2) create copy constructor and do the same thing, but it is more complicated